src/rvoice/fluid_rvoice_event.c

```diff
@@ -94,6 +97,8 @@ static int fluid_rvoice_eventhandler_push_LOCAL(fluid_rvoice_eventhandler_t *han
 
     FLUID_MEMCPY(event, src_event, sizeof(*event));
 
+    fluid_rvoice_eventhandler_notify_pending_events(handler);
```

Comment:
`fluid_rvoice_eventhandler_notify_pending_events` is called from synth context and locks a mutex. This is not so nice. Could we make `handler->idle` an atomic variable instead and avoid the need for the mutex?

Reply:
But isn't `fluid_rvoice_eventhandler_push_LOCAL` the "normal" API side of things and not meant to be called from synthesis context? I was under the impression that the whole point of the rvoice event queue was to separate the normal API from the synthesis context.

However, `fluid_rvoice_eventhandler_update_idle_state()` is definitely called from synthesis context and also uses the muted. But here I tried to make sure that the muted only gets used to signal an idle state. So this only happens if the synth is not doing anything important and therefore wouldn't mind a few extra cycles.

I decided on protecting `handler->idle` with the cond mutex because I wanted to avoid the following race condition:
1. synth context sets handler->idle
2. audio driver notices idle state and calls `fluid_synth_idle_wait`. In idle wait it first ensures that `handler->idle` is 1 and proceeds to call `fluid_cond_wait`.
3. before `fluid_cond_wait` is executed, a new event comes in (in another thread). `notify_pending_events` is called, `handler->idle` is set to 0 and `fluid_cond_signal` is called.
4. audio driver thread continues, actually executes `fluid_cond_wait` and now waits indefinitely (or until the next event, which might never come).

So the goal is to guarantee that `fluid_cond_wait` is only called if the synth is actually still in idle state.

Reply:
Yeah, I meant `fluid_rvoice_eventhandler_update_idle_state` actually. Don't know how I ended up here, must have been very dizzy.

Ok, then pls. add this explanation somewhere in the code. Maybe into the `fluid_rvoice_eventhandler_wait_for_events`. And we can keep the mutex and see how it works out.

Reply:
Will do.